packages/terra-status-view/src/StatusView.jsx

```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Divider from 'terra-divider';
+import styles from './StatusView.scss';
+
+const cx = classNames.bind(styles);
+
+const variants = {
+  NODATA: 'no-data',
+  NOMATCHINGRESULTS: 'no-matching-results',
+  NOTAUTHORIZED: 'not-authorized',
+  ERROR: 'error',
+  NOSERVICE: 'no-service',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * An array of actionable buttons to display.
+   */
+  children: PropTypes.node,
+
+  /**
+   *  Determines if the content should be aligned vertically at the top of the container rather than in the middle.
+   */
+  isAlignedTop: PropTypes.bool,
+
+  /**
+   * Determines if the glyph should be displayed.
+   */
+  isGlyphHidden: PropTypes.bool,
+
+  /**
+   * The descriptive text to display.
+   */
+  message: PropTypes.string,
+
+  /**
+   * The required status title to display. Status views with variants of type no-data, no-matching-results,
+   * not-authorized, and error will have defaulted values unless overridden with this prop.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Determines the type of icon to display.
+   */
+  variant: PropTypes.oneOf([
+    variants.NODATA,
+    variants.NOMATCHINGRESULTS,
+    variants.NOTAUTHORIZED,
+    variants.ERROR,
+    variants.NOSERVICE,
+    variants.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  children: [],
+  isAlignedTop: false,
+  isGlyphHidden: false,
+  message: undefined,
+  title: undefined,
+  variant: variants.ERROR,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class StatusView extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      showGlyph: true,
+      paddingBottom: 0,
+      paddingTop: 0,
+    };
+    this.showAndCenterItems = this.showAndCenterItems.bind(this);
+    this.setInnerNode = this.setInnerNode.bind(this);
+    this.setTitleNode = this.setTitleNode.bind(this);
+    this.setButtonsNode = this.setButtonsNode.bind(this);
+    this.setMessageNode = this.setMessageNode.bind(this);
+    this.setDividerNode = this.setDividerNode.bind(this);
+    this.setGlyphNode = this.setGlyphNode.bind(this);
+  }
+
+  componentDidMount() {
+    // call after all items have rendered in the view to calculate which items
+    // can be shown and center the items inside the view appropriately
+    this.showAndCenterItems();
+  }
+
+  setInnerNode(node) {
+    if (node === null) { return; } // Ref callbacks happen on mount and unmount, element will be null on unmount
```

Comment:
I don't think these checks are necessary. If the component is being unmounted, you probably don't care that your refs are going away. Or were you seeing issues without these?

Reply:
was having trouble of nodes being null when runnings tests at first so switched to this patten seen in the code, fixed the tests so will switch back to providing ref inline when creating the component